pennylane/transforms/defer_measurements.py

```diff
@@ -56,6 +56,16 @@ def _check_tape_validity(tape: QuantumTape):
                 "measurements on a device that does not support them."
             )
 
+    samples_present = any(isinstance(mp, SampleMP) for mp in tape.measurements)
+    postselect_present = any(
+        op.postselect is not None for op in tape.operations if isinstance(op, MidMeasureMP)
+    )
+    if postselect_present and samples_present and tape.batch_size is not None:
+        raise ValueError(
+            "Returning qml.sample is not supported when postselecting mid-circuit "
+            "measurements with broadcasting"
+        )
```

Comment:
Nice!

Reply:
This works
```
dev = qml.device("default.qubit", shots=10)

@qml.qnode(dev)
def f(x):
    for i in range(100):
        qml.RY(x, 0)
        m = qml.measure(0, postselect=1)
    return qml.sample(qml.PauliX(0))

x = np.array([0.4, 0.6])
f(x)
```

```
ValueError: Returning qml.sample is not supported when postselecting mid-circuit measurements with broadcasting
```

Reply:
Could we only raise this error when the arrays are indeed ragged (if that's not too challenging to determine)? We want to give users the option of returning exactly the requested number of shots in Q2, and in that scenario everything should be able to work.

Reply:
I would strongly discourage having the error raised based on whether or not postselection happened to deliver the same number of shots for different batch entries.

Reply:
I agree with @dwierichs . @trbromley we can always come back and update the conditions for this error once the aforementioned feature is implemented. It should be pretty easy for *QNodes* using the transforms. I will have to investigate how easy it might be with just tapes.

Reply:
I agree with @dwierichs. Feels like that would lead to confusion with a user if when they run their program, x times out of 10 they get an error and 10-x times they get a result 🤔